### PR TITLE
feature/create profile

### PR DESCRIPTION
**Pull request by LeaBlancher**
The goal of this PR is to be able to create a user profile and store it in the Firebase database.

It implement:  
- ListProfileViewModel to handle the view of all the profiles, 
- ProfileRepository interface and its implementation using firebase ProfileRepositoryFirebase
- Profile data class with define a profile and its attribute, a section and academic level enum 
- CreateProfileScreen which is the screen needed to create a profile by entering the user info

The actual MainActivty uses the navigation to go to the CreateProfileScreen. When all the information are entered and the user validates its info, it create the profile in the Firebase database and he is redirected to the GreetingScreen (a random screen). 

GreetingScreen will need to be changed to the overview screen or the specific tutor account creation if he choose to be a Tutor.

### CODE REVIEW

**Review by MPepito — APPROVED** [review]
Great job, the implementation looks similar to the model/lesson and mirror the logic, merge and use should not cause any issue I believe.

**Review by alexs-rufer — APPROVED** [review]
This looks perfectly fine to me! great job :)

**Review by MPepito — APPROVED** [review]
Well done, great modularisation and clear implementation, we will have to remember for a later stage to update the average price based on the true average of the lessons
- just the language.student enum typo that could be corrected in a later PR, up to you

File: app/src/main/java/com/github/se/project/model/profile/Profile.kt [hunk 1]
```diff
@@ -52,3 +58,16 @@ enum class AcademicLevel {
   MA4,
   PhD,
 }
+
+enum class Language {
+  STUDENT,
```
[the remark this answers is not in the document]
Comment (LeaBlancher) [reply app/src/main/java/com/github/se/project/model/profile/Profile.kt:63]: It is not a typo and it was already approved and merge to the main so no changes needed.

**Pull request merged**